doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,460 @@
+	Capabilities string `json:"capabilities,omitempty"`
+	// IPAM configuration to be used for this network.
+	IPAM string `json:"ipam,omitempty"`
+	// name of the Linux bridge, if not set will automatically select bridge
+	// based on VF PCI address
+	Bridge string `json:"bridge,omitempty"`
+	// VLAN ID for VF
+	Vlan int `json:"vlan,omitempty"`
+	// VLAN Trunk configuration
+	Trunk []BridgeTrunkConfig `json:"trunk,omitempty"`
+	// enable setting matching vlan tags on the bridge uplink interface, default is false
+	SetUplinkVlan bool `json:"setUplinkVlan,omitempty"`
+	// MTU for VF and representor
+	MTU int `json:"mtu,omitempty"`
+	// MetaPluginsConfig configuration to be used in order to chain metaplugins
+	MetaPluginsConfig string `json:"metaPlugins,omitempty"`
+}
+
+// BridgeTrunkConfig contains configuration for OVS trunk
+type BridgeTrunkConfig struct {
+	MinID *uint `json:"minID,omitempty"`
+	MaxID *uint `json:"maxID,omitempty"`
+	ID    *uint `json:"id,omitempty"`
+}
+
+// BridgeNetworkStatus defines the observed state of BridgeNetwork
+type BridgeNetworkStatus struct {
+}
+
+// BridgeNetwork is the Schema for the ovsnetworks API
+type BridgeNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BridgeNetworkSpec   `json:"spec,omitempty"`
+	Status BridgeNetworkStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// BridgeNetworkList contains a list of BridgeNetwork
+type BridgeNetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BridgeNetwork `json:"items"`
+}
+
+```
+
+#### SriovNetworkNodePolicy CRD
+
+```golang
+// SriovNetworkNodePolicySpec defines the desired state of SriovNetworkNodePolicy
+type SriovNetworkNodePolicySpec struct {
+	// ...existing fields...
+	// contains spec for the software bridge
+	Bridge *Bridge `json:"bridge,omitempty"`
+}
+
+// contains spec for the bridge
+// only one bridge type can be set
+type Bridge struct {
+	// +kubebuilder:validation:Enum=ovs;linux
+	Type string `json:"type"`
+	// contains optional config for OVS bridge
+	Ovs *OVSConfig `json:"ovs,omitempty"`
+	// contains optional config for LinuxBr bridge
+	LinuxBr *LinuxBrConfig `json:"linux,omitempty"`
+}
+
+// OVSConfig optional configuration for OVS bridge and uplink Interface
+type OVSConfig struct {
+	Bridge    *OVSBridgeConfig    `json:"bridge,omitempty"`
+	Interface *OVSInterfaceConfig `json:"interface,omitempty"`
+}
+
+// OVSBridgeConfig contains some options from the Bridge table in OVSDB
+type OVSBridgeConfig struct {
+	DatapathType string            `json:"datapath_type,omitempty"`
+	ExternalIDs  map[string]string `json:"external_ids,omitempty"`
+	OtherConfig  map[string]string `json:"other_config,omitempty"`
+}
+
+// OVSInterfaceConfig contains some options from the Interface table of the OVSDB for PF
+type OVSInterfaceConfig struct {
+	Type        string            `json:"type,omitempty"`
```

Comment:
what values can be provided to type ? (dpdk / netdev ?)

Reply:
yes. I think we should not filter them on the operator level. User may want to use some vendor-specific values for this field.